hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-router/src/main/java/org/apache/hadoop/yarn/server/router/Router.java

```diff
@@ -215,16 +217,48 @@ public void startWepApp() {
 
     Builder<Object> builder =
         WebApps.$for("cluster", null, null, "ws").with(conf).at(webAppAddress);
-    if (RouterServerUtil.isRouterWebProxyEnable(conf)) {
-      fetcher = new FedAppReportFetcher(conf);
-      builder.withServlet(ProxyUriUtils.PROXY_SERVLET_NAME, ProxyUriUtils.PROXY_PATH_SPEC,
-          WebAppProxyServlet.class);
-      builder.withAttribute(WebAppProxy.FETCHER_ATTRIBUTE, fetcher);
-      String proxyHostAndPort = getProxyHostAndPort(conf);
-      String[] proxyParts = proxyHostAndPort.split(":");
-      builder.withAttribute(WebAppProxy.PROXY_HOST_ATTRIBUTE, proxyParts[0]);
+
+    WebAppContext uiWebAppContext = null;
+    boolean isWebUI2Enabled = conf.getBoolean(YarnConfiguration.YARN_WEBAPP_UI2_ENABLE,
+        YarnConfiguration.DEFAULT_YARN_WEBAPP_UI2_ENABLE);
+
+    if(isWebUI2Enabled) {
```

Comment:
Space after the if

Reply:
It may make sense to have this whole chunk under a function.

Reply:
Which can just return the uiWebAppContext

Reply:
Thank you very much for your help to review the code! I will improve this part of the code.